src/Client/FetchClient.php

```diff
@@ -0,0 +1,226 @@
+     */
+    public function __construct(array $config = [])
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return ResponseData
+     */
+    public function __invoke(string $uri, array $config)
+    {
+        $allConfig = $this->mergeAllConfigConfig($config);
+        $client = $this->configureClient($allConfig);
+        $transport = $allConfig['transport']($client);
+        $request = new Request($allConfig['method'], $uri, [], $allConfig['data']);
+
+        return $transport($request);
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return ResponseData
+     */
+    public function get(string $uri, array $config)
+    {
+        return ($this)($uri, $config);
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return ResponseData
+     */
+    public function options(string $uri, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'OPTIONS',
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return ResponseData
+     */
+    public function head(string $uri, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'OPTIONS',
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return ResponseData
+     */
+    public function delete(string $uri, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'DELETE',
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     * @param RequestData $data
+     *
+     * @return ResponseData
+     */
+    public function post(string $uri, mixed $data, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'POST',
+                'data' => $data,
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     * @param RequestData $data
+     *
+     * @return ResponseData
+     */
+    public function put(string $uri, mixed $data, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'PUT',
+                'data' => $data,
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     * @param RequestData $data
+     *
+     * @return ResponseData
+     */
+    public function patch(string $uri, mixed $data, array $config)
+    {
+        return ($this)($uri, merge(
+            $config,
+            [
+                'method' => 'PATCH',
+                'data' => $data,
+            ]
+        ));
+    }
+
+    /**
+     * @param PartialConfig $config
+     *
+     * @return Config
+     */
+    private function mergeAllConfigConfig(array $config): array
```

Comment:
love the naming :smile:

Reply:
Maybe better to suffix it with even more Config's?
What do you think about `mergeAllConfigConfigsConfigConfigurationsConfs`?